### ISSUE TITLE
Customer address EAV attribute does not save or validate on form submission.

### ISSUE THREAD

**Issue by dnadle** [magento/magento2#4613]
## Steps to reproduce

I have tried deploying my project with both 2.0.4 and 2.0.6; same result.
1. Add a customer address EAV attribute of type varchar. I used the following code in UpgradeData for my module:

```
            $customerSetup = $this->_customerSetupFactory->create(['setup' => $setup]);
            $addrEntity = $customerSetup->getEavConfig()->getEntityType('customer_address');
            $attributeSetId = $addrEntity->getDefaultAttributeSetId();

            $attributeSet = $this->_attributeSetFactory->create();
            $attributeGroupId = $attributeSet->getDefaultGroupId($attributeSetId);

            $customerSetup->addAttribute('customer_address', 'shipto', [
                'type' => 'varchar',
                'label' => 'Ship-to Number',
                'input' => 'text',
                'required' => true,
                'visible' => true,
                'user_defined' => true,
                'sort_order' => 1000,
                'position' => 1000,
                'system' => 0
            ]);

            $attribute = $customerSetup->getEavConfig()->getAttribute('customer_address', 'shipto')
            ->addData([
                'attribute_set_id' => $attributeSetId,
                'attribute_group_id' => $attributeGroupId,
                'used_in_forms' => [
                    'adminhtml_customer_address',
                    'customer_address_edit',
                    'customer_register_address',
                ]
            ]);
```
1. Observe that the field appears in the admin customer address edit form as expected.
2. Create a new customer and add an address, making sure to populate the EAV field with a valid string. I used 'TEST'.
## Expected result
1. When the customer is saved with a properly filled address form, the customer record is saved correctly.
## Actual result
1. The UI reports "Ship-to Number" is a required value, and the address is not saved.
## Other results
1. I tried making the EAV attribute not required by modifying the database record. After that, the customer record saves without error, but the EAV attribute value is not saved. When you view the address again the field is blank.

**Comment by dnadle** [comment]
Still a bug in 2.1.0.

**Comment by R4c00n** [comment]
@rgoncharuk Are there any updates on this issue?

**Comment by tuffrabit** [comment]
Also experiencing this. On new account registration, standard address fields are persisted, but custom address attributes are not. This appears to be due to how the custom attribute form fields are submitted, namely as children to 'address'. The standard address fields are not submitted as children to 'address'. Magento\Customer\Controller\Account\CreatePost::extractAddress() is expecting each form field at the root level of the request params array, not nested under something else. As such any custom customer address attributes are not persisted during new customer registration.

**Comment by magento-engcom-team** [comment]
@dnadle, thank you for your report.
The issue is already fixed in 2.2.0
